server/src/main/java/keywhiz/service/permissions/PermissionCheckModule.java

```diff
@@ -8,13 +8,31 @@
 public class PermissionCheckModule extends AbstractModule {
 
   @Override
-  protected void configure() {}
+  protected void configure() {
+    bindToDefaultConstructor(AllowClientSecretCreationPermissionCheck.class);
+  }
 
   @Provides
   public PermissionCheck createPermissionCheck(MetricRegistry metricRegistry,
       AutomationClientPermissionCheck automationClientCheck,
-      OwnershipPermissionCheck ownershipCheck) {
-    PermissionCheck anyPermissionCheck = new AnyPermissionCheck(metricRegistry, List.of(ownershipCheck, automationClientCheck));
+      OwnershipPermissionCheck ownershipCheck,
+      AllowClientSecretCreationPermissionCheck clientSecretCreationCheck) {
+
+    List<PermissionCheck> permissionChecks = List.of(
+        ownershipCheck,
+        automationClientCheck,
+        clientSecretCreationCheck
+    );
+
+    PermissionCheck anyPermissionCheck = new AnyPermissionCheck(metricRegistry, permissionChecks);
```

Comment:
I just realized that the any permission check is short-circuiting, so we'll never be able to see the client secret creation check in action in the logs since, in the current configuration, the automation client check will always pass. I'll re-order these.